ocaml/database/master_connection.ml

```diff
@@ -171,7 +171,10 @@ let open_secure_connection () =
     ~write_to_log:(fun x -> debug "stunnel: %s\n" x)
     ~verify_cert host port
   @@ fun st_proc ->
-  let fd_closed = Thread.wait_timed_read Unixfd.(!(st_proc.Stunnel.fd)) 5. in
+  let polly = Polly.create () in
+  let finally () = Polly.close polly in
+  Fun.protect ~finally @@ fun () ->
+  let fd_closed = Polly.wait polly 1 5000 (fun _ _ _ -> ()) > 0 in
```

Comment:
Good spot, the fd was completely lost here, so Polly was always waiting until a timeout here. It means that it is now no longer able to detect an stunnel connection that dies too quickly.

I should probably add a unit test to detect this in addition to fixing it.

Reply:
Maybe a less error-prone way would be to write an equivalent of `Thread.wait_timed_read` implemented using Polly, instead of select. Then we can unit-test that, and use if everywhere, instead of making subtle mistakes that might go undetected.

Reply:
I've introduced a `wait_timed_read` in Unixext, and replaced all these direct Polly calls with that one.